### PR TITLE
Testing workflows

### PR DESCRIPTION
**Pull request by Electroid**
### What does this PR do?

Do not merge. Work in progress.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

File: packages/bun-internal-test/src/runner.node.mjs [hunk 1]
```diff
@@ -409,6 +409,7 @@ for (const path of allTests) {
         process.exit(1);
       }),
   );
+  break; // HACK
```
Comment (paperclover) [inline packages/bun-internal-test/src/runner.node.mjs]: this isnt intentional for final i assume

File: .github/workflows/build-windows.yml [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+        type: string
+        required: true
+      cpu:
+        type: string
+        required: true
+      assertions:
+        type: boolean
+        default: false
+      deoptimize:
+        type: boolean
```
Comment (paperclover) [inline .github/workflows/build-windows.yml]: is there a reason this cannot be named `optimize` and be a string? with the default being ReleaseFast?

**Review by paperclover — CHANGES_REQUESTED** [review]
i really love how this is coming together.

**Review by paperclover — APPROVED** [review]

File: .github/workflows/build-darwin.yml [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      # TODO: Figure out how to cache homebrew dependencies
+      - name: Install Dependencies
+        env:
+          HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: 1
+          HOMEBREW_NO_AUTO_UPDATE: 1
+          HOMEBREW_NO_INSTALL_CLEANUP: 1
```
Comment (Jarred-Sumner) [inline .github/workflows/build-darwin.yml:119]: we should add that 4th arg

File: .github/workflows/upload.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          run-id: ${{ github.event.workflow_run.id }}
+      - name: Upload to GitHub Releases
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: canary
+          make_latest: false
+          generate_release_notes: false
+          files: bun/*
+      - name: Upload to S3 (using SHA)
+        uses: shallwefootball/s3-upload-action@4350529f410221787ccf424e50133cbc1b52704e
```
Comment (Jarred-Sumner) [inline .github/workflows/upload.yml:33]: I don't love relying on this random github action that hasn't been updated in a year https://github.com/shallwefootball/upload-s3-action

File: .github/workflows/run-test.yml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        timeout-minutes: 30
+        shell: bash
+        env:
+          TMPDIR: ${{ runner.temp }}
+          BUN_FEATURE_FLAG_INTERNAL_FOR_TESTING: "true"
+          SMTP_SENDGRID_SENDER: ${{ secrets.SMTP_SENDGRID_SENDER }}
+          TLS_MONGODB_DATABASE_URL: ${{ secrets.TLS_MONGODB_DATABASE_URL }}
+          TLS_POSTGRES_DATABASE_URL: ${{ secrets.TLS_POSTGRES_DATABASE_URL }}
+        run: |
+          node packages/bun-internal-test/src/runner.node.mjs
```
Comment (Jarred-Sumner) [inline .github/workflows/run-test.yml]:
```suggestion
          node packages/bun-internal-test/src/runner.node.mjs || true
```

**Pull request merged**